### DISCUSSION TITLE
How do I process a the .env for Langchain open ai?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#55535] [Help] [answered]
### Summary

from api/ai.js

```
import { OpenAI } from "langchain/llms/openai";

const model = new OpenAI({ temperature: 0 });

export default model;
```

Error:
Unhandled Runtime Error
Error: OpenAI or Azure OpenAI API key not found

pages\api\ai.js (3:14) @ eval

  1 | import { OpenAI } from "langchain/llms/openai";
  2 | 
> 3 | const model = new OpenAI({ temperature: 0 });
    |              ^
  4 | 
  5 | export default model;

^^^^  I got the same API key working in just Node... but I am trying to integrate it to Next.js with an UI...

The Node code for configuration is just:

```
import { config } from "dotenv";
config();
```

Any help would be awesome!

**Comment by icyJoseph** [comment]
According to this guide, you have to have a `.env` kind of file (there's various variations on that for Next.js):

- https://vercel.com/guides/nextjs-langchain-vercel-ai#getting-started

For example they use `.env.local` to setup the env var, OPENAI_API_KEY

**Comment by Elindo586** [comment]
OK thanks!

I think I need to use express.js type of language for the api page....  and I think the .env is only accessible from the api pages...

I am checking out the template to look at the structure...  thank you much!

**Reply by icyJoseph** [reply, marked as answer]
All you need is a file at the root of your project, like `.env`, Next.js automatically loads those for you.
